### ISSUE TITLE
Phone auth not working with release apk 

### ISSUE THREAD

**Issue by AdlakhaAnkit** [invertase/react-native-firebase#4707]
I have implemented phone based authentication using rnfirebase.

Libraries used:

`"@react-native-firebase/app": "^10.3.0",
    "@react-native-firebase/auth": "^10.3.1",`

Code Snippet:

SignIn Method to send OTP:

`newSignIn = async () => {
        let phoneNumber = this.state.phone;
        if (this.validatePhoneNumber(phoneNumber)) {
          try {
            const confirmation = await auth().signInWithPhoneNumber(phoneNumber);
            ToastAndroid.show(JSON.stringify(confirmation), ToastAndroid.SHORT);
            this.setState({ confirm: confirmation });
          } catch (error) {
            Alert.alert("Sorry!", "Error occurred. Check phone number.");
          }
        } else {
          Alert.alert(
            "Sorry!",
            "Invalid Phone Number, make sure to add + country code."
          );
        }
      }`

When I am testing it on emulator, it is working fine with testing phone numbers. But when I am creating the release apk and installing that apk on my phone then some times it sends OTP and works fine. but some times it is not working, not sending any OTP. I have tried debugging and observed that it is getting into infinite wait on line:

`        const confirmation = await auth().signInWithPhoneNumber(phoneNumber);
`

**Comment by mikehardy** [comment]
Hi there!
Do not mix versions. They should all be the same (10.3.1)
You skipped the template, so I will close this after this comment.

99% of the time people have release mode phone auth problems because they don't configure their SHAs correctly in firebase console.

When having problems, watching adb logcat from Android Studio while a real device is plugged in is typically very instructive

**Comment by AdlakhaAnkit** [comment]
@mikehardy 
I have first tried with same version as 10.3.1 only. Getting this error:


npm ERR! code ETARGET
npm ERR! notarget **No matching version found for @react-native-firebase/app@^10.3.1**
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

Also I have added SHA1 in firebaseconsole. You can suggest what else information needed in issue description insteading f simply closing this.

**Comment by mikehardy** [comment]
The issue template is the suggestion, it should not be ignored in any repo you log issues for, as it puts your issue at risk for pre-emptory closing. If this is not to your liking, open issues and follow templates as a general rule

I see what you mean about app v10.3.1, not sure what happened with that https://www.npmjs.com/package/@react-native-firebase/app - but it's a separate issue, and likely unrelated.
